backend_new/src/controllers/reserved-community-type.controller.ts

```diff
@@ -0,0 +1,100 @@
+import { ReservedCommunityType } from '../dtos/reserved-community-types/reserved-community-type.dto';
+import { ReservedCommunityTypeCreate } from '../dtos/reserved-community-types/reserved-community-type-create.dto';
+import { ReservedCommunityTypeUpdate } from '../dtos/reserved-community-types/reserved-community-type-update.dto';
+import { defaultValidationPipeOptions } from '../utilities/default-validation-pipe-options';
+import { ReservedCommunityTypeQueryParams } from '../dtos/reserved-community-types/reserved-community-type-query-params.dto';
+import { IdDTO } from '../dtos/shared/id.dto';
+import { SuccessDTO } from '../dtos/shared/success.dto';
+
+@Controller('reservedCommunityTypes')
+@ApiTags('reservedCommunityTypes')
+@UsePipes(new ValidationPipe(defaultValidationPipeOptions))
+@ApiExtraModels(
+  ReservedCommunityTypeCreate,
+  ReservedCommunityTypeUpdate,
+  IdDTO,
+  ReservedCommunityTypeQueryParams,
+)
+export class ReservedCommunityTypeController {
+  constructor(
+    private readonly ReservedCommunityTypeService: ReservedCommunityTypeService,
+  ) {}
+
+  @Get()
+  @ApiOperation({ summary: 'List reservedCommunityTypes', operationId: 'list' })
+  @ApiOkResponse({ type: ReservedCommunityType, isArray: true })
+  async list(
+    @Query() queryParams: ReservedCommunityTypeQueryParams,
+  ): Promise<ReservedCommunityType[]> {
+    return await this.ReservedCommunityTypeService.list(queryParams);
+  }
+
+  @Get(`:reservedCommunityTypeId`)
+  @ApiOperation({
+    summary: 'Get reservedCommunityType by id',
+    operationId: 'retrieve',
+  })
+  @ApiOkResponse({ type: ReservedCommunityType })
+  async retrieve(
+    @Param('reservedCommunityTypeId') reservedCommunityTypeId: string,
+  ): Promise<ReservedCommunityType> {
+    return this.ReservedCommunityTypeService.findOne(reservedCommunityTypeId);
+  }
+
+  @Post()
+  @ApiOperation({
+    summary: 'Create reservedCommunityType',
+    operationId: 'create',
+  })
+  @ApiOkResponse({ type: ReservedCommunityType })
+  async create(
+    @Body() reservedCommunityType: ReservedCommunityTypeCreate,
+  ): Promise<ReservedCommunityType> {
+    return await this.ReservedCommunityTypeService.create(
+      reservedCommunityType,
+    );
+  }
+
+  @Put(`:reservedCommunityTypeId`)
+  @ApiOperation({
+    summary: 'Update reservedCommunityType',
+    operationId: 'update',
+  })
+  @ApiOkResponse({ type: ReservedCommunityType })
+  async update(
+    @Body() reservedCommunityType: ReservedCommunityTypeUpdate,
+  ): Promise<ReservedCommunityType> {
+    return await this.ReservedCommunityTypeService.update(
+      reservedCommunityType,
+    );
+  }
+
+  @Delete()
+  @ApiOperation({
+    summary: 'Delete reservedCommunityType by id',
+    operationId: 'delete',
+  })
+  @ApiOkResponse({ type: SuccessDTO })
+  async delete(@Body() dto: IdDTO): Promise<SuccessDTO> {
+    return await this.ReservedCommunityTypeService.delete(dto.id);
```

Comment:
I don't think this endpoint is/will be used so I'm not worried about this issue. When I try to delete the reservedCommunityType that was seeded I run into `Foreign key constraint failed on the field: `listings_reserved_community_type_id_fkey`. That is expected though so I don't think there is anything we need to do about it

Reply:
thats a good call out, I think that if/when we add this to the settings that admins can managed we can add a check for that, but for right now yeah we don't expose the CUD endpoints anywhere